headers/types/dungeon_mode/dungeon.h

```diff
@@ -1635,8 +1652,10 @@ struct dungeon {
     undefined field_0x2ca63;
     undefined field_0x2ca64;
     undefined field_0x2ca65;
-    undefined field_0x2ca66;
-    undefined field_0x2ca67;
+    // 0x02CA66: The cause of the mission over. Identical to the damage source in HandleFaint,
+    // but can be set to some non-damage related reasons manually by the game. IE: "cleared the
+    // dungeon." and "succeeded in the rescue mission."
+    uint16_t fainted_monster_mission_over_reason;
```

Comment:
Wouldn't this be a `enum damage_source_non_move` then?

Reply:
It can be both a `enum damage_source_non_move` or a move_id. Looking again I think I should change it to `union damage_source`?

Reply:
If it can be a move as well, then that would be the best option, yes.